src/arrays/ladderops.jl

```diff
@@ -37,7 +37,16 @@ function momentumop(n::Int)
     return scale(im/sqrt(2.), cop-cop')
 end
 
+squeeze_construct(a::AbstractQuMatrix,b::AbstractQuMatrix, z::Union(Complex64, Float64)) = 0.5*(z'*(a*b)-z*(a'*b'))
```

Comment:
There is a mention of [generalized squeezing operator] (http://qutip.org/docs/3.0.1/apidoc/functions.html#qutip.operators.squeezing) on the QuTip docs. So I thought we could have this, as well we use it in the next construct which is a particular case.